block.js

```diff
@@ -0,0 +1,116 @@
+/**ブロックを表現するクラスで、球との当たり判定をし、ブロックの生存か否かを表している */
+class Block {
+  static HIT_NON = 0;
+  static HIT_LEFT = 1;
+  static HIT_RIGHT = 2;
+  static HIT_TOP = 3;
+  static HIT_BOTTOM = 4;
+
+  static GenerateBlocks(block_row_max) {
+    const block_width = 64;
+    const block_height = 32;
+    const block_colum_max = 10;
+
+    const blockColors = [
+      "red",
+      "blue",
+      "green",
+      "white",
+      "yellow",
+      "aqua",
+      "purple",
+    ];
+
+    let posX = 0;
+    let posY = 0;
+    let color_index = 0;
+    let blocks = [];
+
+    /**
+    y=y軸方向に増えていくことを表している。
+    ｘ＝x軸方向に増えていくことを表している。
+    ｙがrow_max(3列)よりも少ないとき、xがcolum_max(10)列より少ないときブロックが増えることを繰り返す。
+    その際カラーは上記のブロックカラーの配列より割り当てる。カラーは7つなので、超えると0に戻る。
+ */
+    for (let y = 0; y < block_row_max; y++) {
+      for (let x = 0; x < block_colum_max; x++) {
+        let color = blockColors[color_index];
+        color_index++;
+        if (color_index >= blockColors.length) {
+          color_index = 0;
+        }
+
+        let tmp = new Block(posX, posY, color);
+        blocks.push(tmp);
+        posX += block_width;
+      }
+      posY += block_height;
+      posX = 0;
+    }
```

Comment:
コメントが少し微妙かも。
関数など、オブジェクトに対してコメントをするときは、先頭に書くとかでもいいですが、
このように何らかのロジックに対してコメントを書くときは、コメントの下の処理に対する注釈を書いていくべきですね。
あえてめちゃくちゃコメントを書くとすると以下のような感じ。

```suggestion
    // 二重ループで、blockの縦 x 横の階数ループする
    for (let y = 0; y < block_row_max; y++) {
      for (let x = 0; x < block_colum_max; x++) {
        // blockColorsから一つ色を選択して、バラバラのいろを入れる
        let color = blockColors[color_index];
        // 色がばらばらになるようにインクリメントする
        color_index++;
        // color_indexがblockColorsの最後まで行ったらリセットして最初から色をいれていく
        if (color_index >= blockColors.length) {
          color_index = 0;
        }

        // 得られたcolorをもとにBlockインスタンスを生成する
        let tmp = new Block(posX, posY, color);
        // blockに詰める
        blocks.push(tmp);
        // x軸方向の位置を、詰めたブロックの横幅分ずらす
        posX += block_width;
      }
      // x軸方向の位置を、詰めたブロックの縦幅分分ずらす
      posY += block_height;
      // 内部のforから抜けたということは、一行ぶん終わったということなので、xの位置をリセットする
      posX = 0;
    }
```

Reply:
@tachibanayu24 
修正しました！👀　ご確認お願いします👀